src/VirtoCommerce.NotificationsModule.Data/Senders/NotificationSender.cs

```diff
@@ -30,60 +31,35 @@ public class NotificationSender : INotificationSender
             _jobClient = jobClient;
         }
 
-        public async Task ScheduleSendNotificationAsync(Notification notification)
+        public Task ScheduleSendNotificationAsync(Notification notification)
         {
-            if (notification.IsActive.Value)
-            {
-                var message = await CreateMessageAsync(notification);
-
-                _jobClient.Enqueue(() => TrySendNotificationMessageAsync(message.Id));
-            }
+            return CheckAndTryToSendNotificationAsync(notification, true);
         }
 
         [Obsolete("need to use ScheduleSendNotificationAsync")]
         public void ScheduleSendNotification(Notification notification)
         {
             ScheduleSendNotificationAsync(notification).GetAwaiter().GetResult();
         }
-
         
-
-        public async Task<NotificationSendResult> SendNotificationAsync(Notification notification)
+        public Task<NotificationSendResult> SendNotificationAsync(Notification notification)
         {
-            if (notification == null)
-            {
-                throw new ArgumentNullException(nameof(notification));
-            }
-
-            if (notification.IsActive.GetValueOrDefault())
-            {
-                var message = await CreateMessageAsync(notification);
-
-                return await TrySendNotificationMessageAsync(message.Id);
-            }
-
-            return new NotificationSendResult();
+            return CheckAndTryToSendNotificationAsync(notification);
         }
 
-
+        //Must be a public method
```

Comment:
TrySendNotificationMessageAsync(string messageId) is declared as public, that's mean now all previous checks are not executed here?